packages/edit-site/src/components/sidebar-dataviews/default-views.js

```diff
@@ -27,28 +27,9 @@ import {
 } from '../../utils/constants';
 
 export const defaultLayouts = {
-	[ LAYOUT_TABLE ]: {
-		layout: {
-			primaryField: 'title',
-			styles: {
-				title: {
-					maxWidth: 300,
-				},
-			},
-		},
-	},
-	[ LAYOUT_GRID ]: {
-		layout: {
-			mediaField: 'featured_media',
-			primaryField: 'title',
-		},
-	},
-	[ LAYOUT_LIST ]: {
-		layout: {
-			primaryField: 'title',
-			mediaField: 'featured_media',
-		},
-	},
+	[ LAYOUT_TABLE ]: {},
```

Comment:
This PR de-emphasizes what `defaultLayouts` was about. We could make it optional now, as the only cases where consumers would pass this are: 1) they want to disable a specific layout 2) configure grid (badges, or default preview size), or 3) configure table (styles, or default density)

Reply:
It's also about providing the available layouts to switch to. omitting it means no layout switching.

Reply:
I agree we should de-emphasize even more. "badge" should be cross layout. 
I'm not sure how to solve the rest (preview size, styles, default density)

Reply:
**Epiphany:** I feel we may have been thinking about this wrong :) 

Instead of trying to guess how to switch layouts within `DataViews`, it seems the right solution is to actually remove the view type switching logic from the DataViews component itself. It should be just injected as an external component (in header prop maybe)

Reply:
I'm not sure if that's what you're thinking about, but extending layouts is relevant to this conversation. I think the final API could be something along these lines:

```js
<DataViews
  layouts=[ 'table', { type: 'grid', badgeFields: [ 'field' ] }, MyCustomLayoutComponent ]
/>
```

which means:

- use the bundle `table` layout as it is
- use the `grid` layout, and configure the following fields as badge fields
- `MyCustomLayoutComponent` is a new layout that implements everything a layout should have (icon & name for the switcher, behaviour, any view config control, etc.)